### DISCUSSION TITLE
[KC 17.x.x] How to access admin UI from port-forwarded/internal URL / prevent redirect from internal url to KC_HOSTNAME

### DISCUSSION THREAD

**Discussion by armujahid** [keycloak/keycloak#11237] [Ideas]
**Objective:**
keycloak is deployed in kubernetes with proxy=edge with a hostname/KC_HOSTNAME=public.url. I want to restrict admin UI access by allowing it only from internal url by doing port-forwarding

1. kubectl port-forward svc/keycloak 9010:80

1. now http://localhost:9010/admin should work but https://public.url/admin shouldn't work (assuming that I blocked it using reverse proxy)

I have tried setting adminUrl using KC_HOSTNAME_ADMIN but it doesn't work.

**Problem:**
Currently port-forwarded local url http://localhost:9010 is working but it is redirecting me to KC_HOSTNAME :(
As far as I know same setup was working fine with keycloak <17

**Environment variables:**
KC_HOSTNAME=public.url
KC_PROXY=edge

**Related information:**
mentioned at https://www.keycloak.org/docs/latest/server_installation/
"If you do not want to expose the admin endpoints and console on the public domain use the property adminUrl to set a fixed URL for the admin console, which is different to the frontendUrl. It is also required to block access to /auth/admin externally"

**Comment by armujahid** [comment]
I have also posted this at https://keycloak.discourse.group/t/kc-17-x-x-how-to-access-admin-ui-from-port-forwarded-internal-url-prevent-redirect-from-internal-url-to-kc-hostname/14917

**Reply by XcrigX** [reply]
I ran into this same problem on keycloak 18.0.0 (quarkus).  After updating to 18.0.2 (quarkus) it's working for me - with the addition of the KC_HOSTNAME_ADMIN variable.

**Reply by chary1112004** [reply]
Hi @XcrigX , 

After I added ENV KC_HOSTNAME_ADMIN=localhost as your suggestion then there is the page still using public url.

We are using 19.0.1 (quarkus).

![image](https://user-images.githubusercontent.com/76929085/193551098-a3d00648-eea5-4ee4-a6ff-133e3aa8e779.png)

Do you face same issue like this? Any clue in determining the cause of this issue would be very much appreciated.

Thank you!

**Reply by XcrigX** [reply]
I did see something similar - which looked like a likely bug to me. Even when accessing on my intranet using the private KC_HOSTNAME_ADMIN, there was still (at least) one call made to the public hostname when authenticating the admin user. 
I discussed it briefly here: 
https://github.com/keycloak/keycloak/discussions/12466#discussioncomment-2964624

**Comment by stianst** [comment]
Try hostname-admin-url instead
